src/kibana/plugins/settings/sections/objects/_objects.js

```diff
@@ -55,43 +60,82 @@ define(function (require) {
-            service: service.service,
+            service: service.serviceName,
             id: item.id
           };
 
           kbnUrl.change('/settings/objects/{{ service }}/{{ id }}', params);
         };
 
-        $scope.toggleDeleteBtn = function (service) {
-          $scope.deleteAllBtn = _.some(service.data, { checked: true});
+        $scope.bulkDelete = function () {
+          $scope.currentTab.service.delete(_.pluck($scope.selectedItems, 'id')).then(function (resp) {
+            $scope.currentTab.data = _.difference($scope.currentTab.data, $scope.selectedItems);
+            $scope.selectedItems.length = 0;
+          });
         };
 
-        $scope.bulkDelete = function () {
-          var serviceObj = _.find($scope.services, { title: $state.tab });
-          if (!serviceObj) return;
-          var service = $injector.get(serviceObj.service);
-          var ids = _(serviceObj.data)
-            .filter({ checked: true})
-            .pluck('id')
-            .value();
-          service.delete(ids).then(function (resp) {
-            serviceObj.data = _.filter(serviceObj.data, function (obj) {
-              return !obj.checked;
+        $scope.bulkExport = function () {
+          var promises = $scope.selectedItems.map(getExportedItems($scope.currentTab.service));
+          $q.all(promises).then(saveToFile);
+        };
+
+        $scope.exportAll = function () {
+          var promises = $scope.services.reduce(function (promises, service) {
+            return promises.concat(service.data.map(getExportedItems(service.service)));
+          }, []);
+
+          $q.all(promises).then(saveToFile);
+        };
+
+        function getExportedItems(service) {
+          return function (item) {
+            return service.get(item.id).then(function (obj) {
+              return obj.export();
+            });
+          };
+        }
+
+        function saveToFile(results) {
+          var blob = new Blob([angular.toJson(results, true)], {type: 'application/json'});
+          saveAs(blob, 'export.json');
+        }
+
+        $scope.importAll = function (result) {
+          var results;
+          try {
+            results = JSON.parse(result);
+          } catch (e) {
+            return importError();
+          }
+
+          var promises = results.map(function (result) {
+            var service = _.find($scope.services, {type: result._type});
+            if (service == null) return importError();
+
+            return service.service.get().then(function (obj) {
+              return obj.import(result);
             });
-            resetCheckBoxes();
+          });
+
+          $q.all(promises).then(function () {
+            $route.reload();
           });
         };
 
+        function importError() {
+          notify.error('The file could not be processed.');
+        }
+
         $scope.changeTab = function (obj) {
+          $scope.currentTab = _.find($scope.services, {title: obj.title});
```

Comment:
I think we should move the `_.find(...)` into the caller, or accept a string and do the find on that. The only thing that `obj` is used for here is the title, and creates a [weird call here](https://github.com/elastic/kibana/pull/3573/files#diff-244bf15361c0f4a536503fc9ac7ccb5aR36).